src/nvim/os/fs.c

```diff
@@ -192,13 +192,15 @@ int os_open(const char* path, int flags, int mode)
 
 /// Get stat information for a file.
 ///
-/// @return OK on success, FAIL if a failure occurred.
+/// @return `true` on success, `false otherwise.
 static bool os_stat(const char *name, uv_stat_t *statbuf)
   FUNC_ATTR_NONNULL_ALL
 {
   uv_fs_t request;
   int result = uv_fs_stat(uv_default_loop(), &request, name, NULL);
-  *statbuf = request.statbuf;
+  if (result == kLibuvSuccess) {
+    *statbuf = request.statbuf;
+  }
   uv_fs_req_cleanup(&request);
```

Comment:
Hmm. I think it depends on what `uv_fs_req_cleanup(&request)` is doing. If `uv_fs_stat()` succeeds, we set `*statbuf = request.statbuf`. Instead I think we need to copy the buffer, not alias its pointer.